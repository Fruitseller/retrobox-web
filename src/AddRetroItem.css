@import url("https://fonts.googleapis.com/css?family=Montserrat");
* {
  margin: 0;
  padding: 0;
  font-size: 16px;
  font-family: 'Montserrat', sans-serif;
}

body {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  height: 100vh;
  width: 100vw;
  background-color: #f5f6fa;
}

.input {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 600px;
  margin: 0 10px;
}
.input .input-container {
  position: relative;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin: 10px;
}
.input .input-container .input-field {
  padding: 24px;
  background-color: #fff;
  border: none;
  border-radius: 2px;
  width: 100%;
  color: #b7bcdd;
  outline: none;
  z-index: 0;
  -webkit-tap-highlight-color: transparent;
  line-height: 1;
}
.input .input-container .input-field::-webkit-input-placeholder {
  color: #b7bcdd;
  line-height: 1;
  padding-top: 2px;
}
.input .input-container .input-field:hover + .input-field-shadow {
  box-shadow: 0px 2px 10px #d8ddec;
}
.input .input-container .input-field:focus + .input-field-shadow {
  box-shadow: 0px 8px 14px #d8ddec;
  height: 50%;
  width: 97%;
  left: 2%;
}
.input .input-container .input-field-shadow {
  position: absolute;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 99%;
  height: 95%;
  bottom: 0;
  left: 1%;
  z-index: -10;
  box-shadow: 0px 0px 5px #d8ddec;
  border-radius: 2px;
  -webkit-transition: all ease 240ms;
  transition: all ease 240ms;
}
.input .submit-container {
  position: absolute;
  z-index: 0;
  margin: 10px;
  right: -10px;
  top: calc(25% - 11px);
  opacity: 0;
  -webkit-transition: all ease 240ms;
  transition: all ease 240ms;
}
.input .submit-container .submit-btn {
  border: none;
  border-radius: 2px;
  background-color: #94a1f5;
  padding: 12px 20px;
  color: white;
  text-transform: uppercase;
  -webkit-appearance: none;
  -webkit-transition: all ease 240ms;
  transition: all ease 240ms;
  cursor: pointer;
}
.input .submit-container .submit-btn:hover {
  background-color: #7d8df3;
}
.input .submit-container .submit-btn-shadow {
  position: absolute;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  width: 90%;
  height: 50%;
  bottom: 0;
  left: 5%;
  z-index: -5;
  box-shadow: 0px 2px 10px #a3aadb;
  border-radius: 2px;
}
