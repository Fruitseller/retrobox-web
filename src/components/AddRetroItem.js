import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import { withStyles } from 'material-ui/styles';
import Input, { InputLabel } from 'material-ui/Input';
import { FormControl } from 'material-ui/Form';

const styles = theme => ({
  formControl: {
    margin: theme.spacing.unit
  }
});

class AddRetroItem extends React.Component {
  handleSubmit = event => {
    event.preventDefault();
    this.props.addItem(
      this.props.match.params.teamId,
      this.props.uid,
      this.input.value
    );
    this.form.reset();
  };

  render() {
    return (
      <form
        className="form"
        onSubmit={this.handleSubmit}
        ref={form => (this.form = form)}
      >
        <FormControl className={this.props.classes.formControl}>
          <InputLabel htmlFor="thoughts">Enter your thoughts</InputLabel>
          <Input
            id="thoughts"
            inputRef={input => {
              this.input = input;
            }}
          />
        </FormControl>

        <input type="submit" hidden />
      </form>
    );
  }
}

AddRetroItem.propTypes = {
  uid: PropTypes.string.isRequired,
  addItem: PropTypes.func.isRequired
};

export default withStyles(styles)(withRouter(AddRetroItem));
