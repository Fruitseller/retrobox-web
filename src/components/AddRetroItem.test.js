import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import AddRetroItem from './AddRetroItem';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  shallow(
    <AddRetroItem
      uid="unique"
      addItem={() => {}}
      handleAuthentication={() => {}}
      authenticationText="text"
    />
  );
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <AddRetroItem
          uid="unique"
          addItem={() => {}}
          handleAuthentication={() => {}}
          authenticationText="text"
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
