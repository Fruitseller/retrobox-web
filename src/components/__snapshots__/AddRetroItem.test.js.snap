// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="withRouter-RetroAppBar--root-2"
  >
    <header
      className="MuiPaper-root-11 MuiPaper-shadow4-17 MuiAppBar-root-4 MuiAppBar-positionFixed-5 MuiAppBar-colorPrimary-9 mui-fixed"
    >
      <div
        className="MuiToolbar-root-38 MuiToolbar-gutters-39"
      >
        <h2
          className="MuiTypography-root-40 MuiTypography-title-46 MuiTypography-colorInherit-59 MuiTypography-alignCenter-53 withRouter-RetroAppBar--flex-3"
        >
          <a
            className="MuiButtonBase-root-78 MuiButton-root-64 MuiButton-flatContrast-69"
            disabled={false}
            href="/"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiButton-label-66"
            >
              Retrobox
            </span>
            <span
              className="MuiTouchRipple-root-80"
            />
          </a>
        </h2>
        <button
          className="MuiButtonBase-root-78 MuiButton-root-64 MuiButton-flatContrast-69"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label-66"
          >
            text
          </span>
          <span
            className="MuiTouchRipple-root-80"
          />
        </button>
      </div>
    </header>
  </div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root-87 withRouter-AddRetroItem--formControl-1"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <label
        className="MuiFormLabel-root-97 MuiInputLabel-root-91 MuiInputLabel-formControl-92 MuiInputLabel-animated-95"
        data-shrink={false}
        htmlFor="thoughts"
      >
        Enter your thoughts
      </label>
      <div
        className="MuiInput-root-101 MuiInput-formControl-102 MuiInput-inkbar-103 MuiInput-underline-107"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <input
          className="MuiInput-input-110 MuiInput-inputSingleline-113"
          disabled={false}
          id="thoughts"
          onChange={[Function]}
          type="text"
        />
      </div>
    </div>
    <input
      hidden={true}
      type="submit"
    />
  </form>
</div>
`;
